R/bspfGaussianOptimal.R

```diff
@@ -1,8 +1,51 @@
+#' Block Sampling Particle Filter (Linear Gaussian Model; Optimal Proposal).
+#'
+#' @description The \code{blockpfGaussianOpt} function provides a simple example for
+#' \pkg{RcppSMC}. It is based on a block sampling particle filter for a linear
+#' Gaussian model. This is intended only to illustrate the potential of block
+#' sampling; one would not ordinarily use a particle filter for a model in
+#' which analytic solutions are available. The 'optimal' block sampler in the
+#' sense of Doucet, Briers and Senecal (2006) can be implemented in this case.
+#' 
+#' @param data	A vector variable containing the sequence of observations.
+#' 
+#' @param particles	An integer specifying the number of particles.
+#' 
+#' @param lag An integer specifying the length of block to use.
+#' 
+#' @param plot A boolean variable describing whether plot should
+#' illustrate the estimated path along with the uncertainty.
+#' 
+#' @return	The \code{blockpfGaussianOpt} function returns a matrix containing the final
+#' sample paths and a vector containing their weights. The logarithm of the
+#' estimated ratio of normalising constants between the final and initial
+#' distributions is also returned.
+#'
+#' @details The \code{blockpfGaussianOpt} function provides a simple example for
+#' \pkg{RcppSMC}. It is based on a simple linear Gaussian state space model in
+#' which the state evolution and observation equations are:
+#' 	x(n) = x(n-1) + e(n) and 
+#' 	y(n) = x(n) + f(n)
+#' where e(n) and f(n) are mutually-independent standard normal random
+#' variables. The 'optimal' block-sampling proposal described by Doucet
+#' et al (2006) is employed. 
+#' 
+#' @references	A. Doucet, M. Briers, and S. Senecal. Efficient Block Sampling Strategies
+#' for sequential Monte Carlo methods. Journal of Computational and Graphical
+#' Statistics, 15(3):693-711, 2006.
+#' 
+#' @examples
+#' sim <- simGaussian(len=250)
+#' res <- blockpfGaussianOpt(sim$data,lag=5,plot=TRUE)
+#' 
+#' @author		Adam M. Johansen, Dirk Eddelbuettel and Leah F. South
+#' @keywords	programming
+#' @export
 blockpfGaussianOpt <- function(data=c(), particles=1000, lag=5, plot=FALSE) {
 
-    if (length(data == 0)) {
-       #Include some error handling here
-       return;
+    if (length(data) == 0) {
```

Comment:
Given that this is just an illustrative example I guess that this is okay, but I'd shy away from doing this kind of thing in general.

Imagine calling a function within a script which is supposed to do something with your data, but omitting one argument in a long list (it happens to us all...); is it better for the code to fail or to run in an unexpected way producing a line of text in the output which could easily be missed? The answer, of course, depends on the context but I tend to err on the side of paranoia.